pkg/export/setup/setup.go

```diff
@@ -82,9 +83,15 @@ func SetGlobal(exporter *export.Exporter) (err error) {
 // Global returns the global instance of the GCM exporter.
 func Global() *export.Exporter {
 	if globalExporter == nil {
-		// This should usually be a panic but we set an inactive default exporter in this case
-		// to not break existing tests in Prometheus.
+		if !testing.Testing() {
```

Comment:
Ideally we get rid of all use of testing in prod code, this is getting worse and worse. Hinders readability and complexity for testability. Wonder if there are alternatives. How we can test prod code if it does things differently for tests?

Reply:
I agree, but I actually tried to get rid of this and it makes Prometheus unit tests extremely slow by an order of 10x-100x slower. To do this the proper way, it would be far too much code to set the global exporter to a NopExporter in all unit tests.

If you want to use the real exporter in unit tests, simply call `SetGlobal`. This is what `main.go` does in Prometheus.

Reply:
👍🏽 We should be able to remove this part soonish with PRW too.